File: polocloud-suite/src/main/java/dev/httpmarco/polocloud/suite/commands/type/TextArgument.java
package dev.httpmarco.polocloud.suite.commands.type;


import dev.httpmarco.polocloud.suite.commands.CommandArgument;

public final class TextArgument extends CommandArgument<String> {

    public TextArgument(String key) {
        super(key);
    }

    @Override
    public String buildResult(String input) {
        return input;
    }
}

File: agent/src/main/kotlin/dev/httpmarco/polocloud/agent/runtime/local/terminal/arguments/type/TextArgument.kt
package dev.httpmarco.polocloud.agent.runtime.local.terminal.arguments.type

import dev.httpmarco.polocloud.agent.i18n
import dev.httpmarco.polocloud.agent.runtime.local.terminal.arguments.TerminalArgument
import dev.httpmarco.polocloud.agent.runtime.local.terminal.arguments.InputContext

class TextArgument(key: String) : TerminalArgument<String>(key) {

    override fun buildResult(input: String, context: InputContext): String {
        return input
    }

    override fun predication(rawInput: String): Boolean {
        return rawInput.isNotBlank()
    }

    override fun wrongReason(rawInput: String): String {
        return i18n.get("agent.terminal.setup.argument.text.empty")
    }
}
